### ISSUE TITLE
Documentation feedback about some missing or misleading parts

### ISSUE THREAD

**Issue by MaximeBernard** [nrwl/nx#22415]
### Documentation issue

- [ ] Reporting a typo
- [ ] Reporting a documentation bug
- [ ] Documentation improvement
- [x] Documentation feedback

### Additional context or description

I'm in a deep dive nx mode and found out it's missing important (to me!) parts.

### `project.json` or not?
What's the difference between the configuration in `"nx": {}` in the `package.json` and the `project.json`? Is it the same / just a matter of preference? works only in standalone or also in package-based?

### `projectType`
`projectType` could have a bit more documentation about how it's inferred and handled by `nx`. I had to take a look in the `nx` code to make my mind a bit clearer. (in my case they were all inferred as `library` where most of my packages were actual `applications`. 

#### Part 1: How important are they?
From this [doc](https://nx.dev/reference/inputs#named-input-conventions), `nx` is supposed to generate this. 
```json
  "namedInputs": {
    "default": ["{projectRoot}/**/*", "sharedGlobals"], // Default Inputs
    "production": ["default", "!{projectRoot}/jest.config.ts"], // Production Inputs
    "sharedGlobals": [] // Shared Global Inputs
  },
```
Mine (`nx init` in a pre-existing repo) didn't have those lines. Is this important or do you already have this by default in `nx` code anyway? (Figured out when trying `nx repair` out of curiosity).

#### Part 2: Do workspace `NamedInputs` "side-effect" plugins `NamedInputs`?
I didn't find how it's handled. I suppose it's 
`workspace` > `project` > `plugin`
but it wasn't specified. 

So let's say I want to override `default` NamedInput for `eslint` plugin. Should I do that in `workspace` and use it in all my `projects`?

#### Part 3
You may want to merge 
https://nx.dev/reference/nx-json#inputs-namedinputs
in
https://nx.dev/reference/inputs#named-inputs

### Troubleshoot your tasks
You could add a `Troubleshoot your tasks` here https://nx.dev/recipes/troubleshooting.

After running `nx init`, I have 6 `react`/`vite` projects:
- 4 have only `lint` and `test`
- 2 have only `lint` `build` `serve` `preview` and `serve-static`

They are supposed to be "exactly" the same and right now, I'm playing "find the 7 differences" file by file starting by `package.json` with no luck. 

### `project-graph.json`
`project-graph.json` generated in `.nx/cache` helped a lot over my issues. I think it would be worth giving it a proper explanation such as:
- how it's built?
- when?

## Final note

Thank you for taking the time to read this, and I apologize for the lengthy message. I've been following your work for quite some time now and recently decided to join. Last week, @juristr tweeted about Nx being complicated. I thought it would be helpful for you to understand what people (or perhaps just me 😨) find challenging.

**Comment by FrozenPandaz** [comment]
Thank you for your feedback. We'll clarify the parts which you identified.

**Comment by MaximeBernard** [comment]
For the record, the issue I talked about in "Troubleshoot your tasks" has been resolved in https://github.com/nrwl/nx/pull/22618

With more insight now, I think it would be nice to have some `nx graph:debug` (or just some log file actually) where it would verbose everything the build is doing from core to plugin stuff. It might help a lot to understand when things are going wrong.

Unless I missed something, I didn't find how to debug targets and inputs inference.
